broker/features.go

```diff
@@ -1,26 +1,41 @@
 package broker
 
 import (
+	"fmt"
+
 	"github.com/dingotiles/dingo-postgresql-broker/broker/structs"
 	"github.com/frodenas/brokerapi"
+	"github.com/mitchellh/mapstructure"
 )
 
-func (bkr *Broker) clusterFeaturesFromProvisionDetails(details brokerapi.ProvisionDetails) structs.ClusterFeatures {
-	targetNodeCount := defaultNodeCount
-	if rawNodeCount := details.Parameters["node-count"]; rawNodeCount != nil {
-		targetNodeCount = int(rawNodeCount.(float64))
+func (bkr *Broker) clusterFeaturesFromProvisionDetails(details brokerapi.ProvisionDetails) (features structs.ClusterFeatures, err error) {
+	return bkr.clusterFeaturesFromParameters(details.Parameters)
+}
+
+func (bkr *Broker) clusterFeaturesFromUpdateDetails(details brokerapi.UpdateDetails) (features structs.ClusterFeatures, err error) {
+	return bkr.clusterFeaturesFromParameters(details.Parameters)
+}
+
+func (bkr *Broker) clusterFeaturesFromParameters(params map[string]interface{}) (features structs.ClusterFeatures, err error) {
+	err = mapstructure.Decode(params, &features)
+	if err != nil {
+		return
 	}
-	return structs.ClusterFeatures{
-		NodeCount: targetNodeCount,
+	if features.NodeCount < 2 {
+		features.NodeCount = 2
```

Comment:
Yep. I wasn't sure where such a const should go.






On Tue, Jul 5, 2016 at 5:39 PM +1000, "Justin Carter" <notifications@github.com> wrote:












In broker/features.go:
>  	}
> -	return structs.ClusterFeatures{
> -		NodeCount: targetNodeCount,
> +	if features.NodeCount < 2 {
> +		features.NodeCount = 2



Can we extract the magic number '2' into a constant like minimumNodeCount to make it more explicit.



—
You are receiving this because you were assigned.
Reply to this email directly, view it on GitHub, or mute the thread.